### PR TITLE
file-library and items merge

### PR DESCRIPTION
**Pull request by c-ciobanu**
It's almost done, only one little problem that i can't seem to resolve.

When you login and land on the /files route all works fine but if you visit a collection then you come back the api call to retrieve the files fails and the files are not shown.

This is the error message:
screenshot from 2018-12-21 23-23-15
It seems like someone is passing an unwonted sort parameter to the call, I couldn't find where this happens.
screenshot from 2018-12-21 23-25-18
This only happens with certain collections (e.g. kiosks, members, movies).

@rijkvanzanten could you give me a hand with this?

Issue #636

### CODE REVIEW

**Comment by benhaynes** [comment]
Thank you @c-ciobanu!!! @rijkvanzanten is on vacation this week but we'll review (and hopefully merge) this next week! ❤️

File: src/routes/items.vue [hunk 1]
```diff
@@ -238,22 +275,27 @@ export default {
...
+        : this.$route.params.collection;
     },
     emptyCollection() {
-      return (this.meta && this.meta.total_count === 0) || false;
+      return this.meta && this.meta.total_count === 0;
     },
     filters() {
-      if (!this.preferences) return [];
-      return this.preferences.filters || [];
+      return this.preferences && this.preferences.filters
```
Comment (rijkvanzanten) [inline src/routes/items.vue]: I used to prefer smart one-liners over more "dumbed down" solutions, but IMO the old one is easier to read 🙂

Comment (c-ciobanu) [reply src/routes/items.vue]: Yeah, i can see your point, will change them back.

**Comment by rijkvanzanten** [comment]
This is all looking good @c-ciobanu 😄 I haven't been able to figure out why it's not working when you navigate away from some of the other collections though 🤔

**Comment by c-ciobanu** [comment]
@rijkvanzanten just found the problem, it was due to the fact that files is a different route and the `beforeRouteLeave` was not triggering.
Will push this and the other fixes soon.
We should change the name of this file in items-files or something else since i noticed if you try to debug the items file in components it will bug out the debugger and let you see the items file in routes.

**Comment by rijkvanzanten** [comment]
I just pushed another fix for that issue @c-ciobanu 😄 

> We should change the name of this file in items-files or something else since i noticed if you try to debug the items file in components it will bug out the debugger and let you see the items file in routes.

Good point!

**Comment by c-ciobanu** [comment]
@rijkvanzanten had to revert your fix because it was causing another problem on the users page.
Adding a `beforeRouteLeave` resolved it and now all seems to work fine, you should take a look too and see if it's all good.

**Comment by rijkvanzanten** [comment]
There is a big refactor on the horizon that will take care of most of this change, and there have been a bunch of updates on both the item listing and the file listing separately, soo.. While I really really appreciate your efforts @c-ciobanu, I'm afraid I'm going to have to close this as we don't have the time to make it up to date nor be able to get it in an easily maintainable state. Also, I'm not too sure if all the extra conditionals are making things easier, or harder to maintain..

**Pull request closed without merging**